server/jwt_test.go

```diff
@@ -4890,3 +4890,87 @@ func TestJWTResponseThreshold(t *testing.T) {
 		}
 	})
 }
+
+func TestJWTQueuePermissions(t *testing.T) {
+	aExpKp, aExpPub := createKey(t)
+	aExpClaim := jwt.NewAccountClaims(aExpPub)
+	aExpJwt := encodeClaim(t, aExpClaim, aExpPub)
+	newUser := func(t *testing.T, permType string) string {
+		ukp, _ := nkeys.CreateUser()
+		seed, _ := ukp.Seed()
+		upub, _ := ukp.PublicKey()
+		uclaim := newJWTTestUserClaims()
+		uclaim.Subject = upub
+		switch permType {
+		case "allow":
+			uclaim.Permissions.Sub.Allow.Add("foo.> *.dev")
+		case "deny":
+			uclaim.Permissions.Sub.Deny.Add("foo.> *.dev")
+		}
+		ujwt, err := uclaim.Encode(aExpKp)
+		require_NoError(t, err)
+		return genCredsFile(t, ujwt, seed)
+	}
+	confFileName := createConfFile(t, []byte(fmt.Sprintf(`
+		port: -1
+		operator = %s
+		resolver = MEMORY
+		resolver_preload = {
+			%s : %s
+		}`, ojwt, aExpPub, aExpJwt)))
+	defer os.Remove(confFileName)
+	opts, err := ProcessConfigFile(confFileName)
+	if err != nil {
+		t.Fatalf("Received unexpected error %s", err)
+	}
+	errChan := make(chan error, 1)
+	defer close(errChan)
+	s := RunServer(opts)
+	defer s.Shutdown()
+
+	for permType, tests := range map[string][][2]interface{}{
```

Comment:
I find this a bit difficult to read, I think this is easier:
```
for _, test := range []struct{
    permType string
    queue string
    errExpected bool
}{
  {"allow", "queue.dev", false},
  {"allow", "", true},
  {"allow", "bad", false},
  {"deny", "", false},
  {"deny", "queue.dev", true},
} {
  t.Run(test.permType+test.queue, func(t *testing.T) {
```
Then you can use test.queue and test.errExpected instead of getting from the map and doing type casting.

Reply:
done